components/delete_channel_modal/delete_channel_modal.jsx

```diff
@@ -1,38 +1,61 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See License.txt for license information.
 
+import React from 'react';
+import PropTypes from 'prop-types';
+
 import {Modal} from 'react-bootstrap';
-import TeamStore from 'stores/team_store.jsx';
 
 import {FormattedMessage, FormattedHTMLMessage} from 'react-intl';
 
 import {browserHistory} from 'react-router/es6';
 
-import PropTypes from 'prop-types';
+import Constants from 'utils/constants.jsx';
 
-import React from 'react';
+export default class DeleteChannelModal extends React.PureComponent {
+    static propTypes = {
 
-import {deleteChannel} from 'actions/channel_actions.jsx';
-import Constants from 'utils/constants.jsx';
+        /**
+        * Function called when modal is dismissed
+        */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * channel data
+         */
+        channel: PropTypes.object.isRequired,
+
+        /**
+         * Relative Url for team used for redirection after deleting channel
+         */
+        teamRelativeUrl: PropTypes.string.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+            * Function called for deleting channel,
+            */
+
+            deleteChannel: PropTypes.func.isRequired
+        })
+    }
 
-export default class DeleteChannelModal extends React.Component {
     constructor(props) {
         super(props);
 
         this.handleDelete = this.handleDelete.bind(this);
         this.handleKeyDown = this.handleKeyDown.bind(this);
         this.onHide = this.onHide.bind(this);
-
         this.state = {show: true};
     }
 
     handleDelete() {
-        if (this.props.channel.id.length !== 26) {
+        if (this.props.channel.id.length !== Constants.CHANNEL_ID_LENGTH) {
             return;
         }
 
-        browserHistory.push(TeamStore.getCurrentTeamRelativeUrl() + '/channels/town-square');
-        deleteChannel(this.props.channel.id);
+        browserHistory.push(this.props.teamRelativeUrl + '/channels/' + Constants.DEFAULT_CHANNEL);
```

Comment:
I don't get teamUrl from `getCurrentTeamUrl` here.

    export function getCurrentUrl(state) {
        return state.entities.general.credentials.url;
    }

There is no `url` in credentials object. Is there any way I can get baseUrl so I can redirect?